BenchmarkVCFs/SimpleBenchmark.wdl

```diff
@@ -0,0 +1,596 @@
+            header_names = [line.replace('#', '').replace('\n', '') for line in roc if '#' in line][-1].split('\t')
+            df = pd.read_csv(file_path, sep='\t', comment='#', header=None, names=header_names)
+
+            rename_columns = {'score': 'Score', 'true_positives_baseline': 'TP_Base',
+                  'false_positives': 'FP', 'true_positives_call': 'TP_Call', 'false_negatives': 'FN',
+                  'precision': 'Precision', 'sensitivity': 'Recall', 'f_measure': 'F1_Score'}
+
+            df = df.rename(columns=rename_columns)
+            df['Type'] = Type.upper()
+            df['Stratifier'] = "~{strat_label}"
+            df['Score_Field'] = "~{score_field}"
+            df['Call_Name'] = "~{call_output_sample_name}"
+            df['Base_Name'] = "~{base_output_sample_name}"
+
+            roc_summary = pd.concat([roc_summary, df])
+
+        roc_summary.to_csv('ROC_summary.tsv', sep='\t', index=False)
+
+        CODE
+
+    >>>
+
+    runtime {
+        docker: "us.gcr.io/broad-dsde-methods/rtg:v1.0"
+        preemptible: select_first([preemptible, 0])
+        disks: "local-disk " + disk_size + " HDD"
+        cpu: cpu
+        memory: memory + " GB"
+    }
+
+    output {
+        File ROC_summary = "ROC_summary.tsv"
+
+        File tp_base_vcf = "output_dir/tp-baseline.vcf.gz"
+        File tp_base_index = "output_dir/tp-baseline.vcf.gz.tbi"
+        File tp_call_vcf = "output_dir/tp.vcf.gz"
+        File tp_call_index = "output_dir/tp.vcf.gz.tbi"
+        File fp_vcf = "output_dir/fp.vcf.gz"
+        File fp_index = "output_dir/fp.vcf.gz.tbi"
+        File fn_vcf = "output_dir/fn.vcf.gz"
+        File fn_index = "output_dir/fn.vcf.gz.tbi"
+    }
+}
+
+task BCFToolsStats {
+    input {
+        # Post-vcfeval files
+        File tp_base_vcf
+        File tp_base_index
+        File tp_call_vcf
+        File tp_call_index
+        File fp_vcf
+        File fp_index
+        File fn_vcf
+        File fn_index
+
+        String strat_label
+        String bcf_selector
+        String bcf_label
+
+        String call_output_sample_name
+        String base_output_sample_name
+
+        File? evaluation_bed
+
+        Int disk_size = ceil(size(tp_base_vcf, "GB") + size(tp_call_vcf, "GB") + size(fp_vcf, "GB") + size(fn_vcf, "GB")) + 20
+        Int cpu = 16
+        Int memory = 32
+    }
+
+    Int actual_cpu = if cpu < 4 then 4 else cpu    # Make sure to assign at least 4 CPUs to split across 4 parallel procs
+
+    # Handle parsing bcf_selector with surrounding quotes, with empty case handled separately
+    String selection = if bcf_selector!= "" then "-i " + '"' + bcf_selector + '"' else ""
+
+    command <<<
+        set -xeuo pipefail
+
+        # Subset to evaluation bed if provided, to ensure not too many FNs picked up outside of it
+        bcftools stats ~{selection} -s- ~{"-R" + evaluation_bed} --threads ~{floor(actual_cpu / 4)} ~{tp_base_vcf} > tp_base_stats.tsv &
```

Comment:
is there any really advantage to doing it this way, over just multithreading over all the cpus for each call individually?  because if not, this seems like an overcomplification, and will potentially make stdout/stderr all shuffled up in confusing ways.

Reply:
also, if you really want to do it this way, does it makes more sense to use something like parallel?

Reply:
Now that I think about it, I think you're right that there's no good reason to do it this way. In fact, I think this is worse than sequentially doing the processing because some of those files will be smaller than the others, which leads to idle CPUs while waiting for the rest to finish. I changed this to just sequential commands.